.github/actions/build-to-release-branch/action.yml

```diff
@@ -0,0 +1,58 @@
+name: Build to release branch
+description: Merge and build a source branch into a releasable, fully-built branch.
+
+inputs:
+  source_branch:
+    description: Name of source branch.
+    required: true
+  release_branch:
+    description: Name of target release branch. Will be updated with a merge commit including built code.
+    required: true
+  built_asset_paths:
+    description: Generated assets to include in release branch commit. Overrides .gitignore.
+    required: true
+  build_script:
+    description: Command to run to build code. "npm run build" by default.
+    required: false
+
+runs:
+  using: composite
+
+  steps:
+    - name: Configure git
+      shell: bash
+      run: |
+        git config user.name "Your friendly neighborhood GH Actions Bot"
+        git config user.email "<>"
+
+    - name: Check out release branch
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ inputs.release_branch }}
+        fetch-depth: 0
+
+    - name: Merge source into release branch
+      shell: bash
+      # Simulate "-s theirs", which doesn't exist in git for sensible reasons
+      # which do not apply to our current situation.
+      # https://web.archive.org/web/20131126171744/http://www.seanius.net/blog/2011/02/git-merge-s-theirs/
```

Comment:
Sadly this excellent post now exists only in the web archive, and by reference on SO. Entropy, it's a thing.